src/plugins/usage_collection/server/collector/collector_set.ts

```diff
@@ -115,22 +112,40 @@ export class CollectorSet {
       );
     }
 
-    const secondInMs = 1000;
+    const timeoutMs = this.maximumWaitTimeForAllCollectorsInS * SECOND_IN_MS;
     const collectorsWithStatus: CollectorWithStatus[] = await Promise.all(
       [...collectors.values()].map(async (collector) => {
+        const wrappedPromise = performance.timerify(
```

Comment:
I'm wondering: does it make sense to move both `performance.timerify` uses to a function that receives the fn to run and the name to assign the function?

IMO, it'll be easier to understand why we are doing the `Object.defineProperty`, which is essentially creating a named function with dynamic function names :)

WDYT?

Reply:
++